File: 2020/day3/part2/main_test.py
from unittest import TestCase

from advent_of_code.helpers.helpers import read_file, read_file_line_by_line
from advent_of_code.day3.part2.main import slope, main


class TestDay3Part2(TestCase):

    def setUp(self) -> None:
        self.rows = read_file_line_by_line("tests/day3/input_test.txt")

    def test_slope(self) -> None:
        self.assertEqual(
            slope(rows=self.rows, line_skip=1, slide=1), 2
        )
        self.assertEqual(
            slope(rows=self.rows, line_skip=1, slide=3), 7
        )
        self.assertEqual(
            slope(rows=self.rows, line_skip=1, slide=5), 3
        )
        self.assertEqual(
            slope(rows=self.rows, line_skip=1, slide=7), 4
        )
        self.assertEqual(
            slope(rows=self.rows, line_skip=2, slide=1), 2
        )

    def test_main(self) -> None:
        self.assertEqual(
            main("tests/day3/input_test.txt"), 336
        )

File: 2020/day3/part2/main_test.java
package tests.day3.part2;

import advent_of_code.day3.part2.*;
import advent_of_code.helpers.Helpers;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class main_test {
    
    @Test                                               
    @DisplayName("slope")   
    public void testLoop() throws Exception {
        List<String> rows = Helpers.readFileLineByLine("tests/day3/input_test.txt");
        assertEquals(part2.slope(rows, 1, 1), 2);
        assertEquals(part2.slope(rows, 3, 1), 7);
        assertEquals(part2.slope(rows, 5, 1), 3);
        assertEquals(part2.slope(rows, 7, 1), 4);
        assertEquals(part2.slope(rows, 1, 2), 2);
    }

    @Test
    @DisplayName("main")
    public void testMain() throws Exception {
        long count = part2.main("tests/day3/input_test.txt");
        assertEquals(336, count);
    }

}